addons/mail/static/src/discuss/core/common/discuss_notification_settings.xml

```diff
@@ -4,6 +4,14 @@
     <t t-name="mail.DiscussNotificationSettings">
         <div class="o-mail-DiscussNotificationSettings d-flex flex-column">
             <div class="d-flex flex-column my-1">
+                <label class="cursor-pointer d-flex align-items-center my-1">
+                    <h5>Message sound</h5>
+                    <div class="flex-grow-1"/>
+                    <div class="form-check form-switch">
+                        <input class="form-check-input" type="checkbox" role="switch" t-att-checked="store.settings.messageSound" t-on-change="onChangeMessageSound"/>
```

Comment:
Isn't it the definition of t-model? Or there is something else that prevents it in this case?

Reply:
It's basically `t-model`. but `t-model` has some built-in "optimisation" that puts correctness at risk, e.g. some change of value on UI might not necessarily change the actual logical value.

For example here:
https://github.com/odoo/odoo/blob/saas-18.1/addons/mail/static/src/discuss/call/common/call_context_menu.xml#L9

We put a `t-model`, but we saw some corner cases in which the `t-model` fails to sync with UI, therefore we put a `t-on-change` which also kinda the job of `t-model`...

Reply:
The line you linked does have a t-model though, it just has an extra onchange on top of that.

Reply:
Yes it has `t-model`, but I suppose it is irrelevant with the `t-on-change`, and IIRC the `t-on-change` is there because `t-model` doesn't work.